lib/datasource/rubygems/get.ts

```diff
@@ -12,114 +15,166 @@ const INFO_PATH = '/api/v1/gems';
 const VERSIONS_PATH = '/api/v1/versions';
 const DEPENDENCIES_PATH = '/api/v1/dependencies';
 
-export async function getDependencyFallback(
-  dependency: string,
-  registry: string
-): Promise<ReleaseResult | null> {
-  logger.debug(
-    { dependency, api: DEPENDENCIES_PATH },
-    'RubyGems lookup for dependency'
-  );
-  const info = await fetchBuffer<MarshalledVersionInfo[]>(
-    dependency,
-    registry,
-    DEPENDENCIES_PATH
-  );
-  if (!info || info.length === 0) {
-    return null;
+export class InternalRubyGemsDatasource extends Datasource {
+  static readonly id = 'internalrubygems';
+
+  constructor() {
+    super(InternalRubyGemsDatasource.id);
```

Comment:
We should not change the id again. We should pass the `http` reference from caller class to constructor instead.

Changing the id would cause a breaking again, so the id should stay at `rubygems` as it's used for finding auth credentials.

Reply:
Is it sufficient to pass the `id`, or do we want the exact same instance?